syntaxes/noir.tmLanguage.json

```diff
@@ -51,16 +51,64 @@
       ]
     },
     "strings": {
-      "name": "string.quoted.double.nr",
-      "match": "\"(.*?)(\n|(?<!\\\\)\")",
-      "captures": {
-        "1": {
+      "patterns": [
+        {
+          "name": "string.quoted.double.nr",
+          "begin": "\"",
+          "end": "\"",
           "patterns": [
             {
-              "name": "constant.character.escape.nr",
-              "match": "\\\\."
+              "include": "#string-escapes"
             }
           ]
+        },
+        {
+          "name": "string.quoted.double.nr",
+          "begin": "r(#*)\"",
+          "end": "\"(\\1)",
+          "beginCaptures": {
+            "1": {
+              "name": "string.quoted.byte.raw.nr"
+            }
+          },
+          "endCaptures": {
+            "1": {
+              "name": "string.quoted.byte.raw.nr"
+            }
+          }
+        },
+        {
+          "name": "string.interpolated.nr",
+          "begin": "f\"",
+          "end": "\"",
+          "patterns": [
+            {
+              "include": "#interpolated-string-escapes"
+            },
+            {
+              "include": "#interpolations"
+            }
+          ]
+        }
+      ]
+    },
+    "string-escapes": {
+      "name": "constant.character.escape.nr",
+      "match": "\\\\."
+    },
+    "interpolated-string-escapes": {
+      "name": "constant.character.escape.nr",
+      "match": "\\\\.|{{|}}"
+    },
+    "interpolations": {
+      "match": "({)[^\"{}]*(})",
+      "name": "variable.other.nr",
+      "captures": {
+        "1": {
+          "name": "constant.character.nr"
```

Comment:
Ideally we'd use some punctuation thing here. I see other languages use `puncutation.curly.open` or similar, but I can't get to be colored... or, it's colored, but the same way as a variable, maybe, so it doesn't look as nice. I guess for now we can go with this and eventually change it to something better.